pkg/image.go

```diff
@@ -0,0 +1,78 @@
+package skiff
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containers/common/libimage"
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage"
+)
+
+// ImageFromUrl tries to obtain the image from the "most likely source"
+//
+// If the url includes a transport, then the url is parsed and an Image instance
+// is returned.
+//
+// If the url does not contain a transport, then this function first tries to
+// load the image from the (rootless) container storage. If that fails, we try
+// to load it from a registry by prepending the `docker://` transport.
+func ImageFromUrl(url string, ctx context.Context, sysCtx *types.SystemContext) (types.Image, error) {
```

Comment:
context is generally the first argument https://pkg.go.dev/context
```suggestion
func ImageFromUrl(ctx context.Context, sysCtx *types.SystemContext, url string) (types.Image, error) {
```